### PR TITLE
Card registration

### PR DESCRIPTION
**Pull request by Eluminae**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,55 @@ The following gateways are provided by this package:
 For general usage instructions, please see the main [Omnipay](https://github.com/omnipay/omnipay)
 repository.
 
+### Advanced Usage
+
+#### Remembering Cards
```
Comment (everneret) [inline README.md]: Saving cards

File: README.md [hunk 2]
```diff
@@ -36,6 +36,55 @@ The following gateways are provided by this package:
...
+    ...
+));
+$completePurchaseRequest = $gateway->completePurchase($paymentParams);
+$callbackResponse = $completePurchaseRequest->send();
+if ($callbackResponse->hasCreatedCard()) {
+    $cardReference = $callbackResponse->getCardReference();
+}
+```
+
+#### Paying using remembered Cards
```
Comment (everneret) [inline README.md]: saved cards

File: src/Message/PaymentRedirectResponse.php [hunk 3]
```diff
@@ -5,10 +5,7 @@
 use Omnipay\Common\Message\AbstractResponse;
 use Omnipay\Common\Message\RedirectResponseInterface;
 
-/**
- * PayZen Purchase Response
- */
-class PurchaseResponse extends AbstractResponse implements RedirectResponseInterface
+class PaymentRedirectResponse extends AbstractResponse implements RedirectResponseInterface
```
Comment (everneret) [inline src/Message/PaymentRedirectResponse.php]: `PayzenResponse` ?

Comment (Eluminae) [reply src/Message/PaymentRedirectResponse.php]: It's, for me, less clear. It seems like the payzen response while this is instead the redirect to payzen page.

File: src/Message/PurchaseRequest.php [hunk 4]
```diff
@@ -73,6 +54,40 @@ public function getData()
...
+    {
+        return $this->getParameter('cardReference');
+    }
+
+    public function hasCardReference()
+    {
+        return null !== $this->getCardReference();
+    }
+
+    private function guessPageAction()
```
Comment (everneret) [inline src/Message/PurchaseRequest.php]: `definePageAction()` or `resolvePageAction()` ... :thinking:

File: README.md (hunk 1 above)
Comment (Eluminae) [reply README.md]: yep

File: src/Message/PurchaseRequest.php (hunk 4 above)
Comment (Eluminae) [reply src/Message/PurchaseRequest.php]: `getPageAction()` ?

Comment (Eluminae) [reply src/Message/PurchaseRequest.php]: `resolvePageAction` seems ok to me

File: src/Message/CreateCardRequest.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        $data['signature'] = $this->generateSignature($data);
+
+        return $data;
+    }
+
+    public function sendData($data)
+    {
+        return $this->response = new PaymentRedirectResponse($this, $data);
```
Comment (everneret) [inline src/Message/CreateCardRequest.php]: Not fond of calling a payment-related (according to its name) class here, while we're only creating cards.

File: src/Message/PaymentRedirectResponse.php (hunk 3 above)
Comment (everneret) [reply src/Message/PaymentRedirectResponse.php]: `PayzenRedirectResponse` then.

File: src/Message/PurchaseRequest.php (hunk 4 above)
Comment (everneret) [reply src/Message/PurchaseRequest.php]: I was afraid that we mistake this with a simple getter, hence define or resolve. What do you think?

File: src/Message/PaymentRedirectResponse.php (hunk 3 above)
Comment (Eluminae) [reply src/Message/PaymentRedirectResponse.php]: Do you like `redirectToGatewayResponse` ? @everneret

File: src/Message/PurchaseRequest.php (hunk 4 above)
Comment (Eluminae) [reply src/Message/PurchaseRequest.php]: I think there is no simple getter. But resolve seems good. Go for it

File: src/Message/PaymentRedirectResponse.php (hunk 3 above)
Comment (everneret) [reply src/Message/PaymentRedirectResponse.php]: Let's settle for this :wink:

File: src/Message/CreateCardRequest.php (hunk 5 above)
Comment (Eluminae) [reply src/Message/CreateCardRequest.php]: It's now `RedirectToGatewayResponse`

File: src/Message/CompletePurchaseResponse.php [hunk 6]
```diff
@@ -48,4 +48,44 @@ public function getUuid()
...
+            return null;
+        }
+
+        return \DateTime::createFromFormat(
+            'Y-m',
+            sprintf('%s-%s', $this->__data['vads_expiry_year'], $this->__data['vads_expiry_month'])
+        );
+    }
+
+    public function getCardBrand()
```
Comment (Eluminae) [inline src/Message/CompletePurchaseResponse.php]: Or `getCardType`

**Pull request merged**